catalog/data_share_details.go

```diff
@@ -0,0 +1,25 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceShareDetails() *schema.Resource {
+	type sharesData struct {
+		Name      string     `json:"name"`
+		ShareInfo *ShareInfo `json:"share_info,omitempty" tf:"computed"`
+	}
+	return common.DataResource(sharesData{}, func(ctx context.Context, e interface{}, c *common.DatabricksClient) error {
```

Comment:
why not using `ShareInfo` type directly? name is required field there already, so data source semantics will work